tao_compiler/mlir/disc/transforms/lhlo_legalize_roots_to_loops.cc

```diff
@@ -43,18 +47,22 @@ namespace disc_ral {
 namespace {
 
 template <typename LHLO_OpTy>
-LogicalResult elemwiseLowerHelper(
-    OpBuilder& b, Location loc, Operation* op, Value output_linear_index,
-    const ShapeConstraintAnalysis* shape_constraint_analysis, int vector_size) {
+LogicalResult elemwiseLowerHelper(OpBuilder& b, Location loc, Operation* op,
+                                  Value output_linear_index,
+                                  const ShapeAnalysis* shape_analysis,
+                                  int vector_size, LowerConfig* lower_config) {
   if (!isa<LHLO_OpTy>(op) || !op->hasTrait<mlir::OpTrait::Elementwise>())
     return failure();
 
   Value result_memref = cast<lmhlo::LmhloOp>(op).getResultBuffer();
   Value memref = result_memref;
-  if (shape_constraint_analysis) {
-    Value leader_memref =
-        shape_constraint_analysis->GetLeaderValueWithSameShape(result_memref);
-    if (leader_memref != nullptr) memref = leader_memref;
+  if (shape_analysis) {
+    auto fusion = op->getParentOp();
+    if (isa<lmhlo::FusionOp>(fusion)) {
```

Comment:
nit: Why do we need to check if it's inside a fusion here? 
if it' s not inside a fusion, shape_analysis should always be nullptr.

Reply:
In case op's grandparent is the fusion. I can check its grandparent recursively here.

Reply:
Updated. Thanks.